mex.go

```diff
@@ -177,6 +178,7 @@ func (mexset *messageExchangeSet) newExchange(ctx context.Context, framePool Fra
 
 	mexset.exchanges[mex.msgID] = mex
 	mexset.sendChRefs.Add(1)
+	mexset.onAdded()
```

Comment:
the `mexset.mut` is locked. I don't think we want to do that. If we had a peer scoring strategy that accessed the mex, it would probably dead lock at this point. We should have tests that catch this.